enterprise/server/remote_execution/execution_server/execution_server.go

```diff
@@ -771,3 +784,39 @@ func (s *ExecutionServer) PublishOperation(stream repb.Execution_PublishOperatio
 		}
 	}
 }
+
+func (s *ExecutionServer) updateRouter(ctx context.Context, taskID string, executeResponse *repb.ExecuteResponse) error {
+	router := s.env.GetTaskRouter()
+	if router == nil {
+		return nil
+	}
+	// If the result was served from cache, nothing was actually executed, so no
+	// need to update the task router.
+	if executeResponse.GetCachedResult() {
+		return nil
+	}
+	// TODO: Maybe pre-fetch the command at the beginning of PublishOperation,
```

Comment:
you can prolly use readProtoFromCAS here instead

Reply:
Done -- had to attach the user prefix to the context at the beginning of `PublishOperation` to get it to work (I guess when going through the BS API like I was doing before, that happens automatically?). Also cleaned up the `readProtoFromCAS` method a bit -- looks like we can directly call `cachetools.ReadProtoFromCAS` now

Reply:
Yup, makes sense. When you were using the bytestream client before you were going through the external cache impl which does that internally.